database/migrations/TemporaryModels/OptimizeTables.php

```diff
@@ -46,12 +46,58 @@ public function exec(): void
 
 		foreach ($tables as $table) {
 			try {
-				DB::statement($sql . $table);
-				$this->msgSection->writeln('<info>Info:</info> ' . $table . ' optimized.');
+				DB::statement($sql . $table['name']);
 			} catch (\Throwable $th) {
-				$this->msgSection->writeln('<error>Error:</error> could not optimize ' . $table . '.');
+				$this->msgSection->writeln('<error>Error:</error> could not optimize ' . $table['name'] . '.');
 				$this->msgSection->writeln('<error>Error:</error> ' . $th->getMessage());
 			}
 		}
 	}
+
+	/**
+	 * A helper function that allows to drop an index if exists.
+	 *
+	 * @param Blueprint       $table
+	 * @param string|string[] $indexName
+	 */
+	public function dropIndexIfExists(Blueprint $table, string|array $indexName): void
+	{
+		$indexTableName = !is_array($indexName) ? $indexName : ($table->getTable() . '_' . implode('_', $indexName) . '_index');
+		$indexes = collect(Schema::getIndexes($table->getTable()))->map(fn ($a) => $a['name'])->all();
+		if (in_array($indexTableName, $indexes, true)) {
+			$table->dropIndex($indexTableName);
+		}
+	}
+
+	/**
+	 * A helper function that allows to drop an unique constraint if exists.
+	 *
+	 * @param Blueprint $table
+	 * @param string    $indexName
+	 */
+	public function dropUniqueIfExists(Blueprint $table, string $indexName): void
+	{
+		$indexes = collect(Schema::getIndexes($table->getTable()))->map(fn ($a) => $a['name'])->all();
+		if (in_array($indexName, $indexes, true)) {
+			$table->dropUnique($indexName);
+		}
+	}
+
+	/**
+	 * A helper function that allows to drop an foreign key if exists.
+	 *
+	 * @param Blueprint $table
+	 * @param string    $indexName
+	 */
+	public function dropForeignIfExists(Blueprint $table, string $indexName): void
+	{
+		if ($this->driverName === 'sqlite') {
+			return;
+		}
+
+		$fk = collect(Schema::getForeignKeys($table->getTable()))->map(fn ($a) => $a['name'])->all();
+		if (in_array($indexName, $fk, true)) {
+			$table->dropForeign($indexName);
+		}
+	}
```

Comment:
Are these used?

Reply:
Yes in the big migration that Matthias did for refactoring the tables and some other minor migrations. 🙂